### PR TITLE
[KYUUBI #2746][INFRA] Improve NOTICE of binary release

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Maintain notice-binary manually.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @pan3793 @yaooqinn

File: NOTICE-binary [hunk 1]
```diff
@@ -5,3 +5,1667 @@ Copyright 2021-2022 The Apache Software Foundation.
...
+Copyright 2022 The Apache Software Foundation
+
+Spark Project Tags
+Copyright 2022 The Apache Software Foundation
+
+Spark Project Unsafe
+Copyright 2022 The Apache Software Foundation
+
+Spark Project YARN
+Copyright 2022 The Apache Software Foundation
```
Comment (yaooqinn) [inline NOTICE-binary:1663]: do we have these in our binaries?

**Comment by yaooqinn** [comment]
please remove those not belong to the binary releases.

**Comment by ulysses-you** [comment]
@yaooqinn previous package with a wrong command, and now it should be fine

File: build/release/create-package.sh [hunk 2]
```diff
@@ -75,7 +75,7 @@ package_binary() {
 
   echo "Creating binary release tarball ${BIN_TGZ_FILE}"
 
-  ${KYUUBI_DIR}/build/dist --tgz --spark-provided -Pspark-3.2
+  ${KYUUBI_DIR}/build/dist --tgz --spark-provided --flink-provided --hive-provided -Pspark-3.2
```
Comment (ulysses-you) [inline build/release/create-package.sh:78]: the new notice is packaged by this command

**Comment by yaooqinn** [comment]
where is netty?

**Comment by ulysses-you** [comment]
> where is netty?

It seems we lost netty a long time due to the notice is not packaged with netty jar

**Comment by ulysses-you** [comment]
@yaooqinn @pan3793 any comments ? it blocks 1.5.2 release

**Review by pan3793 — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Comment by ulysses-you** [comment]
I will push a backport for banch-1.5 manually

**Pull request closed without merging**